tests/farm1.cpp

```diff
@@ -53,9 +56,9 @@ void farm_example1() {
         [&]() {
             a--; 
             if ( a == 0 ) 
-                return optional<int>(); 
+                return std::experimental::optional<int>(); 
```

Comment:
Declare the lambda as

~~~
[&a]() -> optional<int> {
  a--;
  if (a==0} return {}
  else return a;
}
~~~